cloud/docker/docker_image.py

```diff
@@ -1,457 +1,502 @@
+
 description:
-     - Create, check and remove docker images
+     - Build, load or pull an image, making the image available for creating containers. Also supports tagging an
+       image into a repository and archiving an image to a .tar file.
+
 options:
-  path:
+  archive_path:
     description:
-       - Path to directory with Dockerfile
+      - Use with state 'present' to archive an image to a .tar file.
     required: false
     default: null
-    aliases: []
+  config_path:
+    description:
+      - Path to the Docker CLI config file.
+    default: '~/.docker/config.json'
   dockerfile:
     description:
-       - Dockerfile to use
-    required: false
+      - Use with state 'present' to provide an alternate name for the Dockerfile to use when building an image.
     default: Dockerfile
-    version_added: "2.0"
-  name:
+  force:
     description:
-       - Image name to work with
-    required: true
-    default: null
-    aliases: []
-  tag:
+      - Use with absent state to un-tag and remove all images matching the specified name. Use with states 'present'
+        and 'tagged' to take action even when an image already exists. If archive_path is specified, the force option
+        will cause an existing archive to be overwritten.
+    default: false
+  http_timeout:
     description:
-       - Image tag to work with
+      - Timeout for HTTP requests during the image build operation. Provide a positive integer value for the number of
+        seconds.
+    default: null
     required: false
-    default: "latest"
-    aliases: []
-  nocache:
+  name:
     description:
-      - Do not use cache with building
-    required: false
-    default: false
-    aliases: []
-  docker_url:
+      - "Image name. Name format will be one of: name, repository/name, registry_server:port/name.
+        When pushing or pulling an image the name can optionally include the tag by appending ':tag_name'."
+    required: true
+  path:
     description:
-      - URL of docker host to issue commands to
+      - Use with state 'present' to build an image. Will be the path to a directory containing the context and
+        Dockerfile for building an image.
+    aliases:
+      - build_path
+    default: null
     required: false
-    default: ${DOCKER_HOST} or unix://var/run/docker.sock
-    aliases: []
-  use_tls:
-    description:
-      - Whether to use tls to connect to the docker server.  "no" means not to
-        use tls (and ignore any other tls related parameters). "encrypt" means
-        to use tls to encrypt the connection to the server.  "verify" means to
-        also verify that the server's certificate is valid for the server
-        (this both verifies the certificate against the CA and that the
-        certificate was issued for that host. If this is unspecified, tls will
-        only be used if one of the other tls options require it.
-    choices: [ "no", "encrypt", "verify" ]
-    version_added: "2.0"
-  tls_client_cert:
+  push:
```

Comment:
this could just be infered when 'present' AND name has repository or repository_url could be present as it's own option.